### PR TITLE
Feature/create book entity

### PR DESCRIPTION
**Pull request by Cayman4010**

### CODE REVIEW

File: query
Comment (viacheslav-torbin) [inline query:1]: Why do we need this file?)

**Review by viacheslav-torbin — APPROVED** [review]

File: query
Comment (Cayman4010) [reply query:1]: I don`t know what is it :) It was created by default

File: pom.xml [hunk 1]
```diff
@@ -48,6 +48,24 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-orm</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-dbcp2</artifactId>
+            <version>2.10.0</version>
+        </dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml]: Do you really need all these dependencies?

File: query
Comment (liliia-ponomarenko) [reply query:1]: so, let’s  remove it ;)

File: src/main/java/bookstore/exception/DataProcessingException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package bookstore.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message) {
+        super(message);
+    }
+
+    public DataProcessingException(String s, Exception ex) {
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/exception/DataProcessingException.java]: Do you use this constructor? Maybe you should add some realization;)

File: src/main/java/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/model/Book.java:14]: Lombok is a great library but be [attentive when using it with Entity](https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/)

File: src/main/java/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> getAllBooks =
+                    session.createQuery("FROM Book", Book.class);
+            return getAllBooks.getResultList();
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/repository/impl/BookRepositoryImpl.java]: it can be an inline variable;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Great 😊

**Pull request merged**